pydantic/exceptions.py

```diff
@@ -11,87 +13,48 @@
 )
 
 
-def type_display(type_: type):
-    try:
-        return type_.__name__
-    except AttributeError:
-        # happens with unions
-        return str(type_)
-
-
-Error = namedtuple('Error', ['exc', 'track', 'index'])
+class Error:
```

Comment:
for a smaller performance gain, can we use `__slots__` here.

Reply:
maybe this class isn't required at all and we should just store a tuple, named tuple or dict?